### PR TITLE
Adding support for sysctl

### PR DESCRIPTION
**Pull request by hyperbolic2346**
* Moved kube-proxy config to common layer since now the master uses kube-proxy.
 * Added comments about conntrack settings being handled by kube-proxy and how to set them.
 * Changed rest of code to use leader_get instead of leadership.leader_get since there was a mix before

### CODE REVIEW

**Comment by Cynerva** [comment]
Fixes https://bugs.launchpad.net/charm-kubernetes-worker/+bug/1825436

File: config.yaml [hunk 1]
```diff
@@ -0,0 +1,22 @@
+options:
+  sysctl:
+    type: string
+    default: "{ net.ipv4.neigh.default.gc_thresh1 : 128, net.ipv4.neigh.default.gc_thresh2 : 28672, net.ipv4.neigh.default.gc_thresh3 : 32768, net.ipv6.neigh.default.gc_thresh1 : 128, net.ipv6.neigh.default.gc_thresh2 : 28672, net.ipv6.neigh.default.gc_thresh3 : 32768 }"
+    description: |
+      YAML formatted associative array of sysctl values, e.g.:
+      '{kernel.pid_max : 4194303 }'. Note that kube-proxy handles
+      the conntrack settings. The proper way to alter them is to
+      use the kube-proxy-extra-args config to set them, e.g.:
```
Comment (Cynerva) [inline config.yaml]: kube-proxy-extra-args -> proxy-extra-args

**Comment by Cynerva** [comment]
Doc nitpick aside, this looks good. Thanks.

**Comment by hyperbolic2346** [comment]
> Doc nitpick aside, this looks good. Thanks.

Not a nitpick, it was wrong. I updated the commands but forgot the name above.

**Pull request merged**